[WIP] screen.c cleaning

romgrk opened the pull request:
- Cleanup variables in `win_line()`
 - Refactor `win_line()`

romgrk committed:
use booleans

romgrk committed:
create helper macros

romgrk committed:
screen.c: cleanup win_line() local variables

1. use bools instead of ints where applicable
2. rename vars like: 'attr_pri' => 'attr_has_priority'
3. create vars that increase readability like: 'is_current_window'
4. remove use of 'v', which was used pervasively with no common
   pattern

romgrk committed:
consider enc_utf8 always true

romgrk committed:
extract functions out of win_line

romgrk committed:
remove reference to has_mbyte

romgrk committed:
fix incorrect replacement

romgrk committed:
fix: add field initializers in struct

teto commented on src/nvim/screen.c:
```diff
@@ -5389,66 +5247,41 @@ void screen_puts_len(char_u *text, int textlen, int row, int col, int attr)
-        ScreenLines[off + 1] = ptr[1];
+      }
+      if (mbyte_cells == 2) {
+        ScreenLines[off + 1] = 0;
```
why not ptr[1] anymore ?

romgrk replied:
What is left here is the inner body of the `if (l_enc_utf8)` starting at line 5327 (left). So the `if (mbyte_cells == 2)` here is the one at line 5340 (left), not the replacement of `else if (mbyte_cells == 2)` by `if (mbyte_cells == 2)` as suggested by the highlight.

teto reviewed (approved):
Anything that makes screen.c more readable is gold.
You might want to add braces according to neovim style to please the linter.

romgrk committed:
style & lint

romgrk committed:
extract should_handle_incsearch_highlight()

romgrk committed:
rename variables

tocol        => invert_to
fromcol      => invert_from
fromcol_prev => invert_from_prev
noinvcur     => ! should_invert

romgrk committed:
replace unclear check by booleans

romgrk committed:
move initialization of invert_from/to to declaration

romgrk committed:
cur => current_match

romgrk committed:
int shl_flag => bool shl_flag

romgrk committed:
fix pos.current_match

romgrk committed:
cache conceal_id

romgrk committed:
replace if by ternary

romgrk committed:
replace if by ternary

romgrk committed:
replace variables

lcs_eol_one   => did_print_lcs_eol
lcs_prec_todo => did_print_lcs_prec

romgrk committed:
extract has_match_at_column()

romgrk committed:
extract get_match_attr_at_column()

romgrk committed:
refactor ifs

romgrk committed:
extract get_match_conceal()

romgrk committed:
rename variables

cap_col     => capital_col
capcol_lnum => capital_lnum

romgrk committed:
group spelling initialization code

romgrk committed:
fix missing assignation

romgrk commented on src/nvim/screen.c:
```diff
@@ -2081,6 +2058,151 @@ fill_foldcolumn (
 }
 
+
+static int get_column_diff_until_eol(win_T* wp, int col) {
```
I don't have a clear opinion here, I'll leave it as is unless someone strongly opposes to that name.

justinmk commented:
> Removes references to enc_utf8, enc_dbcs & has_mbyte

How about doing that first, then leave the other changes for a separate PR.

romgrk committed:
extract get_search_match()

romgrk commented:
Sure, I've picked those commit and created a separate PR, I'll base this one on that one: https://github.com/neovim/neovim/pull/6609

justinmk commented:
This should be done in smaller steps, starting from current HEAD. Also it's not necessarily a good idea to create tons of small functions, but factoring out a couple medium-sized ones would be worthwhile.
